### PR TITLE
Pass product name to Boku buy flow (bug 1008419)

### PR DESCRIPTION
**Pull request by kumar303**
@andymckay r?

### CODE REVIEW

File: lib/boku/client.py [hunk 1]
```diff
@@ -234,18 +234,30 @@ def start_transaction(self, callback_url, consumer_id,
...
                 'No multiplier to figure out decimal places for currency {c}'
                 .format(c=currency))
         price = int(price * constants.DECIMAL_PLACES[currency])
+
+        if len(product_name) > 20:
+            log.warning('Truncating product name for Boku. Long name: {n}. '
+                        'Transaction: {t}'.format(n=product_name,
+                                                  t=external_id))
+            product_name = product_name[0:20]
+
```
Comment (andymckay) [inline lib/boku/client.py]: What boku won't truncate it in the ui? bah! len is always a pain for some whacky unicode strings.

File: lib/boku/tests/test_client.py [hunk 2]
```diff
@@ -206,12 +210,29 @@ def test_client_start_transaction_calls_api_with_correct_params(self):
...
+    def test_long_names_are_truncated(self):
+        product_name = 'Unicorn Dust From A Majestic Forest'
+
+        with mock.patch('lib.boku.client.BokuClient.api_call') as mock_client:
+            self.start_transaction(product_name=product_name)
+            eq_(mock_client.call_args[0][1]['desc'],
+                product_name[0:20])
+
+    def test_non_ascii_names(self):
+        product_name = u'Ivan Krsti\u0107'
```
Comment (andymckay) [inline lib/boku/tests/test_client.py]: thanks for the unicode test

**Comment by andymckay** [comment]
r+

File: lib/boku/client.py (hunk 1 above)
Comment (kumar303) [inline lib/boku/client.py]: should we let them truncate it themselves? I don't trust anyone, hah. I guess the main thing is I wanted to put the warning message so maybe I'll just leave that and let Boku truncate.

**Pull request merged**